src/Data/Text/Utf8/AhoCorasick/Automaton.hs

```diff
@@ -31,14 +31,19 @@ module Data.Text.Utf8.AhoCorasick.Automaton where
 
 import Data.Bits (Bits (shiftL, shiftR, (.&.), (.|.)))
 import Data.Char (chr)
+import qualified Data.Char as Char
 import Data.Foldable (foldl')
 import Data.IntMap.Strict (IntMap)
 import qualified Data.IntMap.Strict as IntMap
 import qualified Data.List as List
 import Data.Text.Utf8 (CodeUnit, CodeUnitIndex (CodeUnitIndex), Text (..), indexTextArray)
+import qualified Data.Text.Utf8 as Utf8
 import qualified Data.Vector as Vector
 import qualified Data.Vector.Unboxed as UVector
-import Data.Word (Word64)
+import Data.Word (Word32, Word64)
+
+data CaseSensitivity = CaseSensitive | IgnoreCase
+  deriving Eq
```

Comment:
I think it'd be better to reuse the existing `CaseSensitivity` type. To do this in a backwards compatible way, you can put it into a new shared module, and just re-export it from the other modules that currently export it.